### ISSUE TITLE
False Positives in "Find empty blocks" check

### ISSUE THREAD

**Issue by an-so** [abaplint/abaplint#3610]
## Context
There are two constructs that lead to false positives in the "Find empty blocks" check.

1. There is a new way of using return with an expression, see [ABAP help](https://help.sap.com/doc/abapdocu_latest_index_htm/latest/en-US/abapreturn.htm).
2. Certain EML statements.

### return [expr]
The new return [expr] construct causes "Find empty blocks" false positives, if the only statement in the block ist a return [expr].

Examples:
```
IF cond.
  RETURN some_class=>some_method_call().
ELSE.
  RETURN some_class=>some_other_method_call().
ENDIF.

IF sy-subrc = 0.
  RETURN abap_true.
ENDIF.
``` 

### EML statements
Here, it is not entirely clear which constructs/statements are affected. Examples are

```
LOOP AT x REFERENCE INTO DATA(y) WHERE <condition>.

  MODIFY ENTITIES OF a IN LOCAL MODE
                 ENTITY ent
                 UPDATE FIELDS (f) WITH VALUE #(...).

ENDLOOP.


CASE x.
  WHEN value1.
    SELECT a
    FROM b
    WHERE cond
    ORDER BY sort ASCENDING, nr ASCENDING
    INTO TABLE tab.

  WHEN value2.
    ...

ENDCASE.


IF cond.
  COMMIT ENTITIES x OF y 
    FAILED DATA(z). 
ENDIF.
```

## Additional Information
Currently, we are using abaplint-cli version "2.105.15". I cannot check if it occurs on newer versions as well.

**Comment by an-so** [comment]
@larshp Sure, I added my company. I am a colleague of @agoeb's :)

Unfortunately, changing the version is not that easy in my context. I thought, maybe you knew by heart if such a bug had already been fixed. Or otherwise, you could test it rather easy.
